chatsky/pipeline/pipeline/component.py

```diff
@@ -40,40 +40,56 @@ class PipelineComponent(abc.ABC, BaseModel, extra="forbid", arbitrary_types_allo
-    :param timeout: (for asynchronous only!) Maximum component execution time (in seconds),
-        if it exceeds this time, it is interrupted.
-    :param requested_async_flag: Requested asynchronous property;
-        if not defined, `calculated_async_flag` is used instead.
-    :param calculated_async_flag: Whether the component can be asynchronous or not
-        1) for :py:class:`~.pipeline.service.service.Service`: whether its `handler` is asynchronous or not,
-        2) for :py:class:`~.pipeline.service.group.ServiceGroup`: whether all its `services` are asynchronous or not.
-
-    :param start_condition: StartConditionCheckerFunction that is invoked before each component execution;
-        component is executed only if it returns `True`.
-    :type start_condition: Optional[:py:data:`~.StartConditionCheckerFunction`]
-    :param name: Component name (should be unique in single :py:class:`~.pipeline.service.group.ServiceGroup`),
-        should not be blank or contain `.` symbol.
-    :param path: Separated by dots path to component, is universally unique.
     """
 
     before_handler: BeforeHandler = Field(default_factory=BeforeHandler)
+    """
+    :py:class:`~.BeforeHandler`, associated with this component.
+    """
     after_handler: AfterHandler = Field(default_factory=AfterHandler)
+    """
+    :py:class:`~.AfterHandler`, associated with this component.
+    """
     timeout: Optional[float] = None
+    """
+    (for asynchronous only!) Maximum component execution time (in seconds),
+    if it exceeds this time, it is interrupted.
+    """
     requested_async_flag: Optional[bool] = None
+    """
+    Requested asynchronous property; if not defined,
+    `calculated_async_flag` is used instead.
+    """
     calculated_async_flag: bool = False
+    """
+    Whether the component can be asynchronous or not.
+
+    1) for :py:class:`~.pipeline.service.service.Service`: whether its `handler` is asynchronous or not,
+    2) for :py:class:`~.pipeline.service.group.ServiceGroup`: whether all its `services` are asynchronous or not.
+
+    """
     start_condition: StartConditionCheckerFunction = Field(default=always_start_condition)
+    """
+    StartConditionCheckerFunction that is invoked before each component execution;
+    component is executed only if it returns `True`.
+    """
     name: Optional[str] = None
+    """
+    Component name (should be unique in single :py:class:`~.pipeline.service.group.ServiceGroup`),
+    should not be blank or contain `.` symbol.
+    """
     path: Optional[str] = None
+    """
+    Separated by dots path to component, is universally unique.
+    """
 
     @model_validator(mode="after")
-    def pipeline_component_validator(self):
-        if self.name is not None and (self.name == "" or "." in self.name):
-            raise Exception(f"User defined service name shouldn't be blank or contain '.' (service: {self.name})!")
+    def __pipeline_component_validator(self):
```

Comment:
Add doc to validators: need to document when they raise and what data they add support for (i.e. validation from `Callable` for services).